api/envoy/extensions/internal_redirect/allowlisted_routes/v3/allowlisted_routes_config.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package envoy.extensions.internal_redirect.allowlisted_routes.v3;
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.internal_redirect.allowlisted_routes.v3";
+option java_outer_classname = "AllowlistedRoutesConfigProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Allowlisted routes internal redirect predicate]
+
+// An internal redirect predicate that accepts only explicitly allowed target routes.
+// [#extension: envoy.internal_redirect_predicates.allowlisted_routes]
+message AllowlistedRoutesConfig {
+  // The list of routes that's allowed as redirect target by this predicate,
+  // identified by the route's :ref:`name <envoy_api_field_config.route.v3.Route.route>`.
+  repeated string allowed_route_names = 1;
```

Comment:
empty names allowed?

Reply:
Good catch. Added validation to disallow empty route name here.